apps/web/src/pages/settings/WebhookPage/useWebhookPage.ts

```diff
@@ -0,0 +1,100 @@
+import { useClipboard } from '@mantine/hooks';
+import { successMessage, errorMessage } from '@novu/design-system';
+import { IResponseError, checkIsResponseError } from '@novu/shared';
+import { MAIL_SERVER_DOMAIN, useEnvController, updateDnsSettings } from '@novu/shared-web';
+import { useMutation } from '@tanstack/react-query';
+import { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { validateMxRecord } from '../../../api/inbound-parse';
+import { useEffectOnce } from '../../../hooks';
+import { getWebhookClaimStatusFromEnvironment } from './getWebhookClaimStatusFromEnvironment';
+import { WebhookClaimStatus } from './WebhookPage.types';
+
+export const useWebhookPage = () => {
+  const mailServerDomain = `10 ${MAIL_SERVER_DOMAIN}`;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** ‏This was copied from the previous implementation... should we do anything better here?‏